### PR TITLE
Improve SSL approach

### LINKED ISSUE #4942: Updates to the Pages module to set the default value for "Secure" to be `true` when the site is configured for SSL Enabled & Enforced

### PR DESCRIPTION
**Pull request by donker**
This PR aims to do the following:

1. Simplify the UI that manages the SSL settings for a site
2. Introduce a simplified "make this site SSL" setting that doesn't require any other tweaks and ensures all content can only be accessed through https.

The overarching goal is to improve security. Users can get lost in the various settings and mistakes are easily made.

## Implementation

The changes are as follows:

1. Introduction of a new portal setting SSLSetup. This setting is an enum which (for now) consists of "Off", "On" and "Advanced". This setting replaces the old SSLEnabled setting.
2. The SSL settings page in the Personabar now starts with a dropdown that sets the above setting. The other bits of the UI are shown dependent on the settings.
3. Where necessary the code is adjusted to ensure that a site set to SSLSetup "On" always assumes URLs must be SSL.

Resolves #4942

### CODE REVIEW

**Comment by bdukes** [comment]
I've updated the base for this PR to the new `release/9.11.0` branch, it is ready to merge once approved

**Review by bdukes — APPROVED** [review]

File: Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Components/Security/Ssl/PortalStats.cs
Comment (mitchelsellers) [inline Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Components/Security/Ssl/PortalStats.cs:20]: I believe that this field, and the following 2 should be either renamed to "NumberOfTabs" or "TabCount" to provide clear, non-shorted variable names.

**Review by mitchelsellers — CHANGES_REQUESTED** [review]
Looks good, and great behavior changes I think this will help.  I did make one suggestion for naming/consistency.

**Review by valadas — APPROVED** [review]
Other than the mentioned suggestions and a double-check from @bdukes on potential breaking changes, this looks great to me. Nice one!

File: DNN Platform/Library/Entities/Portals/PortalSettings.cs
Comment (bdukes) [reply DNN Platform/Library/Entities/Portals/PortalSettings.cs:656]: I don't think this should cause an issue. We'd had an issue with `PortalId` delegating to `PortalID` like this, but it was the casing issue, not the property usage.

**Review by valadas — APPROVED** [review]
Thank you sir.

**Review by mitchelsellers — APPROVED** [review]

**Comment by WillStrohl** [comment]
I'm late to the party, but I have two questions... 

1. Shouldn't the primary underlying `SSLEnabled` property in the API be marked for deprecation and removed later?  
2. Shouldn't this setting and the web service be superuser-only?  An administrator may not have the necessary knowledge/expertise to know what happens turning this feature on and off.

**Pull request merged**
